src/box/applier.cc

```diff
@@ -344,7 +344,11 @@ applier_connect(struct applier *applier)
 			      &applier->addr_len);
```

Comment:
Agree. The factory concept looks rather ugly. I'll try to switch to the def/ctx concept, as you suggested. I'll post a comment when it's ready. Thanks!

Reply:
I reworked the patch as you suggested - now instead of `iostream_factory` I use `iostream_ctx` without any virtual methods. PTAL.

Reply:
Here's a brief summary of what's changed:

1. Renamed `iostream::ctx` to `iostream::data` to avoid confusion with `iostream_ctx` (iostream: rename ctx to data).
2. Reworked the patch that added `iostream_factory` (iostream: introduce context and use it for creating streams).
3. Rebased the patch that added SSL stub (Add SSL iostream stub).

Other commits haven't been changed.